### PR TITLE
TI_Matching Query for Azure SQL and Azure Key Vault

### PR DESCRIPTION
**Pull request by aprakash13**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+id: 57c7e832-64eb-411f-8928-4133f01f4a25
+name: TI map IP entity to Azure Key Vault logs
+description: |
+  'Identifies a match in Azure Key Vault logsfrom any IP IOC from TI'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ThreatIntelligence
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml:7]: We should add the connector for Azure Key Vault Logs here as well, or do we not have one?

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - ThreatIntelligenceIndicator
+  - connectorId: ThreatIntelligenceTaxii
+    dataTypes:
+      - ThreatIntelligenceIndicator
+queryFrequency: 1h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml:18]: Is there no technique we can map to?

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  | where KeyVaultEvents_TimeGenerated  >= TimeGenerated and KeyVaultEvents_TimeGenerated < ExpirationDateTime
+  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+  | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, Url, ExpirationDateTime, ConfidenceScore, KeyVaultEvents_TimeGenerated,
+  TI_ipEntity, ClientIP, ResourceId, SubscriptionId, OperationName, ResultType, CorrelationId, id_s, clientInfo_s, httpStatusCode_d, identity_claim_appid_g, identity_claim_http_schemas_microsoft_com_identity_claims_objectidentifier_g
+  | extend timestamp = KeyVaultEvents_TimeGenerated
+  entityMappings:
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPCustomEntity
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureKeyVault.yaml:45]: I don't see IPCustomEntity projected in the query.

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+id: d0aa8969-1bbe-4da3-9e76-09e5f67c9d85
+name: TI map IP entity to Azure SQL Security Audit Events
+description: |
+  'Identifies a match in SQLSecurityAuditEvents from any IP IOC from TI'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ThreatIntelligence
+    dataTypes:
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml:8]: We should add the connector for Azure Key Vault Logs here as well, or do we not have one?

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      - ThreatIntelligenceIndicator
+  - connectorId: ThreatIntelligenceTaxii
+    dataTypes:
+      - ThreatIntelligenceIndicator
+queryFrequency: 1h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Impact
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml:18]: Is there no technique we can map to?

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  | extend timestamp = SQLSecurityAuditEvents_TimeGenerated
+  entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: AccountCustomEntity
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IPCustomEntity
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml:50]: I don't see IPCustomEntity projected in the query.

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  | where SQLSecurityAuditEvents_TimeGenerated  >= TimeGenerated and SQLSecurityAuditEvents_TimeGenerated < ExpirationDateTime
+  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+  | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, Url, ExpirationDateTime, ConfidenceScore, SQLSecurityAuditEvents_TimeGenerated,
+  TI_ipEntity, ClientIP, LogicalServerName_s, action_name_s, database_name_s, object_name_s, additional_information_s, application_name_s, host_name_s, server_principal_name_s, succeeded_s, database_principal_name_s, NetworkIP,  NetworkDestinationIP, NetworkSourceIP, EmailSourceIpAddress
+  | extend timestamp = SQLSecurityAuditEvents_TimeGenerated
+  entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: FullName
+        columnName: AccountCustomEntity
```
Comment (petebryan) [inline Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml:46]: I don't see AccountCustomEntity projected in the query.

**Review by petebryan — CHANGES_REQUESTED** [review]

File: Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml (hunk 5 above)
Comment (shainw) [reply Detections/ThreatIntelligenceIndicator/IPEntity_AzureSQL.yaml:18]: We don't do technique's for IOC matching, we decided early on with Jason Wescott that we would leave it off.  Happy to revisit again, but okay for now.

**Review by petebryan — APPROVED** [review]

**Pull request merged**